include/os/freebsd/spl/sys/debug.h

```diff
@@ -142,10 +160,97 @@ spl_assert(const char *buf, const char *file, const char *func, int line)
 		if (unlikely(!(0 == _verify0_right)))			\
 		    spl_panic(__FILE__, __FUNCTION__, __LINE__,		\
 		    "VERIFY0P(" #RIGHT ") "				\
-		    "failed (NULL == %p)\n",				\
+		    "failed (NULL == %px)\n",				\
 		    (void *)_verify0_right);				\
 	} while (0)
 
+/*
+ * Note that you should not put any operations you want to always happen
+ * in the print section unless you only want them to run on debug builds!
+ * e.g. VERIFY3Uf(2, <, 3, "%s", foo(x)), foo(x) won't run on non-debug
```

Comment:
Should this be ASSERT3Uf? VERIFY3Uf is always expanded, even on non-debug builds

Reply:
It's been a minute since I wrote this, but I certainly think so, I can't see a way this as written can be accurate.